### PR TITLE
Bump abacus to 1.8.14 and various isolated fixes.

### PR DESCRIPTION
**Pull request by prashanDYDX**
Screenshot 2024-07-01 at 12 00 42 PM

Screenshot 2024-07-01 at 12 01 26 PM

### CODE REVIEW

File: v4/feature/receipt/src/main/java/exchange/dydx/trading/feature/receipt/components/leverage/DydxReceiptPositionLeverageView.kt [hunk 1]
```diff
@@ -104,7 +105,7 @@ object DydxReceiptPositionLeverageView : DydxComponent {
                 } else {
                     null
                 },
-                direction = PlatformDirection.from(state.after?.leverage, state.before?.leverage),
+                direction = PlatformDirection.from(state.after?.leverage?.absoluteValue, state.before?.leverage?.absoluteValue),
```
Comment (prashanDYDX) [inline v4/feature/receipt/src/main/java/exchange/dydx/trading/feature/receipt/components/leverage/DydxReceiptPositionLeverageView.kt:108]: fixes short position leverage changes

File: v4/feature/trade/src/main/java/exchange/dydx/trading/feature/trade/margin/components/liquidationprice/DydxAdjustMarginInputLiquidationPriceView.kt [hunk 2]
```diff
@@ -117,24 +117,28 @@ object DydxAdjustMarginInputLiquidationPriceView : DydxComponent {
             ) {
                 Spacer(modifier = Modifier.weight(1f))
                 if (state.before != null) {
+                    val beforeTextStyle = if (state.after == null) {
```
Comment (prashanDYDX) [inline v4/feature/trade/src/main/java/exchange/dydx/trading/feature/trade/margin/components/liquidationprice/DydxAdjustMarginInputLiquidationPriceView.kt:120]: if there is a before and after, before text is slightly smaller according to designs.

File: v4/platformUI/src/main/java/exchange/dydx/platformui/components/buttons/PlatformButton.kt [hunk 3]
```diff
@@ -104,11 +105,10 @@ fun PlatformButton(
                 leadingContent()
             }
             if (text != null) {
-                Text(
-                    style = TextStyle.dydxDefault.themeFont(fontSize = fontSize)
+                PlatformAutoSizingText(
```
Comment (prashanDYDX) [inline v4/platformUI/src/main/java/exchange/dydx/platformui/components/buttons/PlatformButton.kt:107]: Target leverage didn't fit when it was 4 digits (ex: 13.58x) which comes up when initializing to existing position leverage

File: v4/platformUI/src/main/java/exchange/dydx/platformui/components/textgroups/PlatformAutoSizingText.kt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+                drawContent()
+            }
+        },
+        style = scaledTextStyle,
+        softWrap = false,
+        maxLines = 1,
+        onTextLayout = { textLayoutResult ->
+            if (textLayoutResult.hasVisualOverflow) {
+                scaledTextStyle =
+                    scaledTextStyle.copy(fontSize = scaledTextStyle.fontSize * 0.9)
```
Comment (mike-dydx) [inline v4/platformUI/src/main/java/exchange/dydx/platformui/components/textgroups/PlatformAutoSizingText.kt]: will this effectively recursively reduce by 10% until it fits?

**Review by mike-dydx — APPROVED** [review]

File: v4/platformUI/src/main/java/exchange/dydx/platformui/components/textgroups/PlatformAutoSizingText.kt (hunk 4 above)
Comment (prashanDYDX) [reply v4/platformUI/src/main/java/exchange/dydx/platformui/components/textgroups/PlatformAutoSizingText.kt]: yup exactly, weirdly/unfortunately there doesn't seem to be an existing component/property that computes exact size to fit

**Pull request merged**